### ISSUE TITLE
Gnome Pomodoro no longer builds on Fedora Rawhide (F30)

### ISSUE THREAD

**Issue by mbooth101** [codito/gnome-pomodoro#402]
We just rebuilt the whole distro due to rebasing on glibc 2.29 and gcc 9.0.

I found that gnome pomodoro no longer builds. Here are the full logs:

Build root installation log: https://kojipkgs.fedoraproject.org//work/tasks/4679/32394679/root.log
Build log: https://kojipkgs.fedoraproject.org//work/tasks/4679/32394679/build.log

I suppose the interesting bits are as follows...

(From the build root installation:)

```
DEBUG util.py:492:   dbus-devel                       i686    1:1.12.12-2.fc30         build   38 k
DEBUG util.py:492:   dbus-glib-devel                  i686    0.110-3.fc29             build   70 k
DEBUG util.py:492:   desktop-file-utils               i686    0.23-9.fc29              build   70 k
DEBUG util.py:492:   gettext                          i686    0.19.8.1-18.fc30         build  1.0 M
DEBUG util.py:492:   glib2-devel                      i686    2.59.1-1.fc30            build  471 k
DEBUG util.py:492:   gnome-common                     noarch  3.18.0-6.fc29            build  130 k
DEBUG util.py:492:   gnome-desktop3-devel             i686    3.31.4-1.fc30            build   63 k
DEBUG util.py:492:   gobject-introspection-devel      i686    1.59.3-1.fc30            build  1.2 M
DEBUG util.py:492:   gom-devel                        i686    0.3.3-7.fc29             build   47 k
DEBUG util.py:492:   gstreamer1-devel                 i686    1.15.1-2.fc30            build  540 k
DEBUG util.py:492:   gtk3-devel                       i686    3.24.4-1.fc30            build  4.6 M
DEBUG util.py:492:   intltool                         noarch  0.51.0-13.fc29           build   57 k
DEBUG util.py:492:   libappindicator-gtk3-devel       i686    12.10.0-22.fc30          build   20 k
DEBUG util.py:492:   libappstream-glib-devel          i686    0.7.14-4.fc30            build  149 k
DEBUG util.py:492:   libcanberra-devel                i686    0.30-17.fc29             build   31 k
DEBUG util.py:492:   libnotify-devel                  i686    0.7.7-6.fc29             build   29 k
DEBUG util.py:492:   libpeas-devel                    i686    1.22.0-9.fc29            build  103 k
DEBUG util.py:492:   sqlite-devel                     i686    3.26.0-1.fc30            build  145 k
DEBUG util.py:492:   telepathy-glib-devel             i686    0.24.1-9.fc29            build  1.2 M
DEBUG util.py:492:   vala                             i686    0.43.6-2.fc30            build  1.4 M
```

(From the build:)
```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I/builddir/build/BUILD/gnome-pomodoro-0.14.0 -I/builddir/build/BUILD/gnome-pomodoro-0.14.0/lib -DGETTEXT_PACKAGE=\"gnome-pomodoro\" -DPACKAGE_LOCALE_DIR=\"/usr/share/locale\" -DPACKAGE_DATA_DIR=\"/usr/share/gnome-pomodoro\" -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0 -I/usr/include/libdrm -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gsettings-desktop-schemas -I/usr/include/gobject-introspection-1.0 -I/usr/include/libpeas-1.0 -pthread -I/usr/include/gom-1.0 -w -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -c resources.c  -fPIC -DPIC -o .libs/libgnome_pomodoro_la-resources.o
BUILDSTDERR: application.vala:149.43-149.71: error: Assignment: Cannot convert from `GLib.List<Gtk.Window>' to `GLib.List<weak Gtk.Window>?'
BUILDSTDERR:             unowned List<weak Gtk.Window> windows = this.get_windows ();
BUILDSTDERR:                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
BUILDSTDERR: application.vala:151.20-151.26: error: The name `windows' does not exist in the context of `Pomodoro.Application.get_last_focused_window'
BUILDSTDERR:             return windows != null
BUILDSTDERR:                    ^^^^^^^
BUILDSTDERR: application.vala:152.23-152.29: error: The name `windows' does not exist in the context of `Pomodoro.Application.get_last_focused_window'
BUILDSTDERR:                     ? windows.first ().data
BUILDSTDERR:                       ^^^^^^^
BUILDSTDERR: application.vala:69.9-69.30: error: struct `Pomodoro.Application.Options' cannot be empty
BUILDSTDERR:         private struct Options
BUILDSTDERR:         ^^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 4 error(s), 0 warning(s)
```

I hope this means more to you than it does to me!

**Comment by jbicha** [comment]
I believe this is specifically triggered by vala 0.43/0.44.

**Comment by codito** [comment]
Build is fixed with b4dea40.
